java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -807,27 +807,23 @@ private void enqueueUncheckedExceptions() {
   }
 
   private boolean isUncheckedExceptionCatchBlock(BytecodeCFG.Block b) {
-    String exceptionTypeName = b.getExceptionType();
-    if (exceptionTypeName == null) {
-      return false;
-    }
-    Type exceptionType = semanticModel.getClassType(exceptionTypeName);
-    return ExceptionUtils.isUncheckedException(exceptionType);
+    return b.isCatchBlock() && ExceptionUtils.isUncheckedException(b.getExceptionType(semanticModel));
   }
 
   private ProgramState stateWithException(ProgramState programState, BytecodeCFG.Block b) {
-    Type exceptionType = semanticModel.getClassType(b.getExceptionType());
+    Type exceptionType = b.getExceptionType(semanticModel);
     SymbolicValue.ExceptionalSymbolicValue sv = new SymbolicValue.ExceptionalSymbolicValue(exceptionType);
     return programState.stackValue(sv);
   }
 
   private void enqueueExceptionHandlers(Type exceptionType, ProgramState ps) {
     programPosition.block.successors().stream()
       .map(b -> (BytecodeCFG.Block) b)
-      .filter(b -> b.getExceptionType() != null)
+      .filter(BytecodeCFG.Block::isCatchBlock)
       .filter(b -> b.isUncaughtException()
         ||/*required as long as there is no real type tracking*/ exceptionType == null
-        || exceptionType.isSubtypeOf(b.getExceptionType()))
+        || exceptionType.isSubtypeOf(b.getExceptionType(semanticModel))
```

Comment:
I would try to spare two consecutive calls to `b.getExceptionType(semanticModel)`. Dedicated small method?